netbout-web/src/main/java/com/netbout/dynamo/DyMessages.java

```diff
@@ -202,7 +203,11 @@ public long unread() throws IOException {
             )
             .where(DyFriends.HASH, Conditions.equalTo(this.bout))
             .where(DyFriends.RANGE, Conditions.equalTo(this.self))
-            .iterator().next();
+            .iterator();
+        if (!iterator.hasNext()) {
+            throw new Inbox.BoutNotFoundException(this.bout);
```

Comment:
@original-brownbear i agree. should we accept `HttpException` as default behaviour? if it's so, integration test seems like a not big issue.